src/plugins/expandOperation.ts

```diff
@@ -0,0 +1,98 @@
+import { Host, Session } from "@azure-tools/autorest-extension-base";
+import { CodeModel, Request, Operation, Parameter } from "@azure-tools/codemodel";
+import { isNullOrUndefined } from "util";
+import { Helper } from "../helper";
+import { CliConst, CliCommonSchema } from "../schema";
+import { NodeHelper } from "../nodeHelper";
+import { Modifier } from "./modifier/modifier";
+import { CopyHelper } from "../copyHelper";
+
+export class ExpandOperation{
+
+    constructor(protected session: Session<CodeModel>){
+    }
+
+    public async process() {
+
+        await this.modifier();
+
+        for (const group of this.session.model.operationGroups) {
+            const existedNames = new Set<string>(group.operations.map((op) => op.language.default.name.toLowerCase()));
+            const expandedGroupOperations = [];
+            for (const operation of group.operations) {
+                const expandNames = NodeHelper.getCliExpandOperationNames(operation);
+                if (!expandNames) {
+                    continue;
+                }
+                const expandedOperations = this.expandOperations(expandNames, operation, existedNames);
+                
+                expandedOperations.forEach((expandedOperation) => {
+                    // Link src operation to expanded operation
+                    NodeHelper.addCliOperation(operation, expandedOperation);
+                    // Link expaneded operation to src opreation
+                    NodeHelper.setExpandOperationOriginalOperation(expandedOperation, operation);
+    
+                    expandedGroupOperations.push(expandedOperation);
+                });
+
+                if (expandedOperations.length > 0) {
+                    NodeHelper.setHidden(operation, true);
+                }
+            }
+            expandedGroupOperations.forEach((op) => group.addOperation(op));
+        }
+    }
+
+    private async modifier() {
+        const directives = await this.session.getValue(CliConst.CLI_EXPAND_OPERATION_DIRECTIVE_KEY, null);
+        if (directives) {
+            Helper.dumper.dumpCodeModel('modifier-pre');
```

Comment:
>modifier-pre [](start = 41, length = 12)

it's better to have the plug-in scope here, we have another step called modifier and just calling it modifier will make things confusing. and, this one would be the same as the dump you do out side, isn't it?

Reply:
add prefix `split-operation-` to distinguish with other plugin